src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter
+            TIME_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    public static final int NAME = 1;
+    public static final int PERIOD = 0;
+    public static final int HOURS = 2;
+    public static final int SALARY = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom;
+        LocalDate localDateTo;
+        LocalDate localDateData;
+
+        try {
+            localDateFrom = LocalDate.parse(dateFrom, TIME_FORMATTER);
+            localDateTo = LocalDate.parse(dateTo, TIME_FORMATTER);
+        } catch (DateTimeParseException exc) {
+            throw new RuntimeException("Period format for scanning is not valid", exc);
+        }
+        StringBuilder builder = new StringBuilder();
```

Comment:
Let's name your variables more informative, same in all places
```suggestion
        StringBuilder report = new StringBuilder();
```